map-storage/src/MapStorageServer.ts

```diff
@@ -0,0 +1,86 @@
+        callback: sendUnaryData<EditMapCommandMessage>
+    ): void {
+        const editMapCommandMessage = call.request.editMapCommandMessage;
+        if (!editMapCommandMessage || !editMapCommandMessage.editMapMessage?.message) {
+            callback(null, { id: "", editMapMessage: undefined });
+            return;
+        }
+        const gameMap = mapsManager.getGameMap(call.request.mapKey);
+        if (!gameMap) {
+            // TODO: Send an error?
+            callback(null, { id: editMapCommandMessage.id, editMapMessage: undefined });
+            return;
+        }
+        const editMapMessage = editMapCommandMessage.editMapMessage.message;
+        try {
+            switch (editMapMessage.$case) {
+                case "modifyAreaMessage": {
+                    const message = editMapMessage.modifyAreaMessage;
+                    const area = gameMap.getGameMapAreas().getArea(message.id, AreaType.Static);
+                    if (area) {
+                        const areaObjectConfig: ITiledMapRectangleObject = {
+                            ...area,
+                            ...message,
+                        };
+                        mapsManager.executeCommand(call.request.mapKey, {
+                            type: "UpdateAreaCommand",
+                            areaObjectConfig,
+                        });
+                    }
+                    break;
+                }
+                case "createAreaMessage": {
+                    const message = editMapMessage.createAreaMessage;
+                    console.log(message);
+                    const areaObjectConfig: ITiledMapRectangleObject = {
+                        ...message,
+                        visible: true,
+                        class: "area",
+                    };
+                    mapsManager.executeCommand(call.request.mapKey, {
+                        areaObjectConfig,
+                        type: "CreateAreaCommand",
+                    });
+                    break;
+                }
+                case "deleteAreaMessage": {
+                    const message = editMapMessage.deleteAreaMessage;
+                    console.log("DELETE AREA MESSAGE HANDLED");
+                    mapsManager.executeCommand(call.request.mapKey, {
+                        type: "DeleteAreaCommand",
+                        id: message.id,
+                    });
+                    break;
+                }
+                default: {
+                    throw new Error(`UNKNOWN EDIT MAP MESSAGE CASE. THIS SHOULD NOT BE POSSIBLE`);
+                }
+            }
+            // send edit map message back as a valid one
+            callback(null, editMapCommandMessage);
+        } catch (e) {
+            console.log(e);
+            // do not send editMap message back?
+            callback(null, { id: editMapCommandMessage.id, editMapMessage: undefined });
```

Comment:
Maybe we should send the error back instead? (or a stringified version of the error?)